test/e2e/capi_test.go

```diff
@@ -259,3 +261,78 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 		})
 	}
 })
+
+var _ = Describe("When upgrading a workload cluster [K8s-Upgrade]", func() {
```

Comment:
Is this spec deviating from other specs too much to have a separate `Describe` instead of being another `context`?

Reply:
it's not, I should be able to merge them. I just wanted to avoid running the upgrades as part of the regular capi job but I can add a GINGKO_SKIP for that

Reply:
https://github.com/kubernetes/test-infra/pull/24365/files